### PR TITLE
Fix QgsNetworkAccessManager ssl error handler race condition issue

### PR DESCRIPTION
**Pull request by troopa81**
Use the same logic than with auth request handler #45833

It fixes randomly failing test test_core_networkmanager ([here](https://github.com/qgis/QGIS/actions/runs/3143795872/jobs/5111110008) for instance)

I tested it with several iterations and random sleep in the code. 

Even if I'm quite confident this won't break anything, I'm not keen on backporting it

### CODE REVIEW

File: src/core/network/qgsnetworkaccessmanager.cpp [hunk 1]
```diff
@@ -434,16 +427,17 @@ void QgsNetworkAccessManager::onReplySslErrors( const QList<QSslError> &errors )
...
   // while in worker thread the signal will be queued (and return immediately) -- hence the need to lock the thread in the next block
   emit sslErrorsOccurred( reply, errors );
   if ( this != sMainNAM )
   {
     // lock thread and wait till error is handled. If we return from this slot now, then the reply will resume
     // without actually giving the main thread the chance to act on the ssl error and possibly ignore it.
-    mSslErrorHandlerMutex.lock();
-    mSslErrorWaitCondition.wait( &mSslErrorHandlerMutex );
-    mSslErrorHandlerMutex.unlock();
+    mSslErrorHandlerSemaphore.acquire();
```
Comment (strk) [inline src/core/network/qgsnetworkaccessmanager.cpp:439]: It's already acquired outside the conditional block if I read correctly

**Comment by nyalldawson** [comment]
Thanks @troopa81 -- this has been an issue for a loooong time :+1:

File: src/core/network/qgsnetworkaccessmanager.cpp (hunk 1 above)
Comment (troopa81) [reply src/core/network/qgsnetworkaccessmanager.cpp:439]: Yes, that's the point! You can read more in this [PR](https://github.com/qgis/QGIS/pull/45833), that's kind of the same issue

The goal is to block until the semaphore is released in *afterAuthRequestHandled*

Comment (strk) [reply src/core/network/qgsnetworkaccessmanager.cpp:439]: Thanks, it's really not intuitive to understand two lines being .acquire() and .release(), maybe add detailed comments to explain the rationale for that ?

**Review by nyalldawson — APPROVED** [review]

**Review by strk — APPROVED** [review]
Thanks

**Pull request merged**